### PR TITLE
OEL-164: Drupal 9 compatibility fixes.

### PR DESCRIPTION
**Pull request by escuriola**
### Description

Drupal 9 compatibility fixes

### Change log

- Added: core version 8 || 9, other additions
- Changed:
- Deprecated:
- Removed: require-dev dependencies from composer
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: oe_contact_forms.install [hunk 1]
```diff
@@ -10,9 +10,9 @@ declare(strict_types = 1);
 /**
  * Implements hook_install().
  */
-function oe_contact_forms_install() {
+function oe_contact_forms_install($is_syncing) {
   // Configure the RDF SKOS graphs.
-  if (!\Drupal::service('config.installer')->isSyncing()) {
+  if ($is_syncing) {
```
Comment (drishu) [inline oe_contact_forms.install:15]: I think you inverted the logic here

Comment (escuriola) [reply oe_contact_forms.install:15]: Ups, you're right. I'm going to fix now and review

File: composer.json [hunk 2]
```diff
@@ -6,26 +6,24 @@
...
-        "php": ">=7.2"
+        "openeuropa/rdf_skos": "1.x",
+        "openeuropa/oe_corporate_countries": "dev-EWPP-914",
+        "php": ">=7.3"
     },
     "require-dev": {
-        "composer/installers": "~1.5",
-        "drupal/core-composer-scaffold": "^8.8",
-        "drupal/coder": "8.3.8",
+        "composer/installers": "^1.11",
```
Comment (nagyad) [inline composer.json:18]: This can remain with ~ instead of ^.

File: composer.json [hunk 3]
```diff
@@ -6,26 +6,24 @@
...
         "drupal/config_devel": "~1.2",
+        "drupal/core-composer-scaffold": "^8.9 || ^9.1",
         "drupal/drupal-extension": "~4.0",
-        "drush/drush": "~9.7@stable",
-        "guzzlehttp/guzzle": "~6.3",
-        "openeuropa/code-review": "~1.0.0-beta2",
-        "openeuropa/drupal-core-require-dev": "^8.9.13",
-        "openeuropa/task-runner": "~1.0.0-beta5",
-        "phpunit/phpunit": "~6.0"
+        "drush/drush": "^10.3",
```
Comment (nagyad) [inline composer.json:22]: This can remain with ~ instead of ^.

File: composer.json [hunk 4]
```diff
@@ -6,26 +6,24 @@
...
+        "openeuropa/code-review": "^1.6",
```
Comment (nagyad) [inline composer.json:23]: This can remain with ~ instead of ^.

File: tests/src/Functional/ContactFormSanitizeCommandTest.php [hunk 5]
```diff
@@ -84,22 +89,22 @@ public function testContactFormMessageDataSanitization() {
...
-    $this->assertEqual($sanitized_message->get('oe_telephone')->value, '+000-' . $message_id);
-    $this->assertEqual($sanitized_message->get('oe_topic')->value, 'topic-' . $message_id);
+    $this->assertEquals($sanitized_message->get('name')->value, 'User' . $message_id);
+    $this->assertEquals($sanitized_message->get('mail')->value, 'user+' . $message_id . '@example.com');
+    $this->assertEquals($sanitized_message->get('subject')->value, 'subject-by-' . $message_id);
+    $this->assertEquals($sanitized_message->get('message')->value, 'message-by-' . $message_id);
+    $this->assertEquals($sanitized_message->get('ip_address')->value, '127.0.0.1');
+    $this->assertEquals($sanitized_message->get('oe_country_residence')->target_id, 'residence-in-' . $message_id);
+    $this->assertEquals($sanitized_message->get('oe_telephone')->value, '+000-' . $message_id);
+    $this->assertEquals($sanitized_message->get('oe_topic')->value, 'topic-' . $message_id);
```
Comment (nagyad) [inline tests/src/Functional/ContactFormSanitizeCommandTest.php:99]: assertEquals first parameter is the expected and second is the actual.

File: tests/src/Functional/ContactFormSanitizeCommandTest.php [hunk 6]
```diff
@@ -84,22 +89,22 @@ public function testContactFormMessageDataSanitization() {
...
-    $this->assertEqual($plain_message->get('name')->value, $data['name']);
-    $this->assertEqual($plain_message->get('mail')->value, $data['mail']);
-    $this->assertEqual($plain_message->get('subject')->value, $data['subject']);
-    $this->assertEqual($plain_message->get('message')->value, $data['message']);
-    $this->assertEqual($plain_message->get('ip_address')->value, $data['ip_address']);
+    $this->assertEquals($plain_message->get('name')->value, $data['name']);
+    $this->assertEquals($plain_message->get('mail')->value, $data['mail']);
+    $this->assertEquals($plain_message->get('subject')->value, $data['subject']);
+    $this->assertEquals($plain_message->get('message')->value, $data['message']);
+    $this->assertEquals($plain_message->get('ip_address')->value, $data['ip_address']);
```
Comment (nagyad) [inline tests/src/Functional/ContactFormSanitizeCommandTest.php:107]: assertEquals first parameter is the expected and second is the actual.

File: tests/src/Kernel/AccessTest.php [hunk 7]
```diff
@@ -128,7 +128,7 @@ public function testContactBlockAccess(): void {
     $definition = $block_manager->getDefinition($plugin_id);
 
     // Assert we have the contact form block definition.
-    $this->assertEqual($definition['id'], 'oe_contact_forms_corporate_block');
+    $this->assertEquals($definition['id'], 'oe_contact_forms_corporate_block');
```
Comment (nagyad) [inline tests/src/Kernel/AccessTest.php:131]: assertEquals first parameter is the expected and second is the actual.

File: tests/src/Kernel/BlocksTest.php [hunk 8]
```diff
@@ -62,9 +62,9 @@ public function testBlockDerivatives(): void {
     $build = $plugin->build();
 
     // Assert we have the correct build.
-    $this->assertEqual($build['header']['#markup'], $header);
-    $this->assertEqual($build['privacy_policy']['#title'], $privacy_text);
-    $this->assertEqual($build['oe_topic']['widget']['#title'], $topic_label);
+    $this->assertEquals($build['header']['#markup'], $header);
+    $this->assertEquals($build['privacy_policy']['#title'], $privacy_text);
+    $this->assertEquals($build['oe_topic']['widget']['#title'], $topic_label);
```
Comment (nagyad) [inline tests/src/Kernel/BlocksTest.php:67]: assertEquals first parameter is the expected and second is the actual.

**Review by nagyad — CHANGES_REQUESTED** [review]

**Review by nagyad — DISMISSED** [review]

**Review by sergepavle — CHANGES_REQUESTED** [review]
Also on rerun of upgrade status tests I've found that we still have few issues which it would be better to fix.
```FILE:
modules/custom/oe_contact_forms/tests/src/FunctionalJavascript/MessageFormTest.p
hp

STATUS         LINE                           MESSAGE                           
--------------------------------------------------------------------------------
Fix now        197  Call to deprecated method assertUrl() of class              
                    Drupal\Tests\BrowserTestBase. Deprecated in drupal:8.2.0 and
                    is removed from drupal:10.0.0. Use                          
                    $this->assertSession()->addressEquals() instead.            
--------------------------------------------------------------------------------
Fix now        221  Call to deprecated method assertUrl() of class              
                    Drupal\Tests\BrowserTestBase. Deprecated in drupal:8.2.0 and
                    is removed from drupal:10.0.0. Use                          
                    $this->assertSession()->addressEquals() instead.            
--------------------------------------------------------------------------------
Fix now        281  Call to deprecated method assertUrl() of class              
                    Drupal\Tests\BrowserTestBase. Deprecated in drupal:8.2.0 and
                    is removed from drupal:10.0.0. Use                          
                    $this->assertSession()->addressEquals() instead.            
--------------------------------------------------------------------------------
```

**Review by sergepavle — APPROVED** [review]

**Review by nagyad — DISMISSED** [review]

**Review by nagyad — APPROVED** [review]

**Review by yenyasinn — APPROVED** [review]

**Pull request merged**